_episodes/11.md

```diff
@@ -0,0 +1,252 @@
+└── ...
+{% endhighlight %}
+
+When querying for data under January 1st, 1992, according to the Hive model,
+query engines like Hive and Trino will only scan orc files under the 
+`orders/orderdate=1992-01-01` directory. The idea is to avoid scanning
+unnecessary data by grouping rows based on a field commonly used in a query.
+
+In episode 6/7, we discussed a bit about how a query gets represented internally
+to Trino once you submit your SQL query. First, the Parser will convert SQL to
+an abstract syntax tree (AST) format. Then the Planner generates a different
+ tree structure called the intermediate represenation (IR) that contains nodes 
+representing the steps that need to be performed in order to answer the query. 
+The leaves of the tree get executed first, and the parents of each node are 
+dependent on the action of its child completing before it can start. Finally, 
+the planner/cost-based-optimizer (CBO) will run various updates on the IR to
+optimize the query plan until it is ready to be executed. To sum it all up, the 
+planner and CBO generate and optimize the plan by running optimization rules. 
+Refer to chapter 4 in 
+[Presto: The Definitive Guide](https://www.starburstdata.com/oreilly-presto-guide-download/)
+pg. 50 for more information.
+
+In episode 9, we discussed how hash-joins work by first drawing a nested-loop
+analogy to how joins work. We then discussed how it is advantageous to read the
+inner loop into memory to avoid a lot of extra disk calls. Since it is ideal to
+read an entire table into memory, you will likely want to make sure the table
+that is built in memory is the smaller size of the two tables. This smaller
+table called the build table. The table that gets streamed is called the probe
+table. We discussed a bit how hash-joins work which is a common mechanism to
+execute joins in a distributed and parallel fashion.  
+
+Another nomenclature akin to build table and probe tables are dimension and
+fact table, respectively. This nomenclature comes from the [star schema](https://en.wikipedia.org/wiki/Star_schema)
+that comes from data warehousing. Typically, there are large tables called fact
+tables that would live at the center of the schema. These tables typically have
+many foreign keys, and a bit of quantitative or measuarable columns of the event
+or instance. The foreign keys connect these big fact tables to smaller dimension
+tables that when joined provide human readable context to enrich the recordings
+in the fact table. The schema ends up looking like a star with the fact table at
+the center. In essence, you just need to remember when someone is describing a
+fact table they are saying it is a bigger table that is likely going to end up
+on the probe side of a join, where a dimension is more likely a candidate to fit
+into memory on the build side of a join.
+
+So let's get onto the Dynamic Filtering shall we?
+
+Dynamic Filtering takes advantage of joins with big fact tables to smaller
```

Comment:
Make it specific that it's node local dynamic filtering?

Reply:
Can do!

Reply:
So actually my intent was to talk about the generic premise of dynamic filters and the last sentence I allude to looking at the local dynamic filtering blog for more understanding on how that works. Then talk a bit about how local is different from dynamic partition pruning.

Reply:
Updated this